internal/git/output_parser.go

```diff
@@ -86,6 +87,7 @@ func parseLog(out io.Reader, maxBytes int) []*Commit {
 
 		scanner.Scan()
 		currentCommit.AuthorEmail = string(scanner.Bytes()[2:])
+		emails = append(emails, currentCommit.AuthorEmail)
```

Comment:
Hmm I'm not the biggest fan of this approach. We are already returning the commits array so why return data twice (emails list)? Architecturally this also opens up the possibility of these lists containing different emails which doesn't make sense for the consumer. I would typically solve something like this with a utility function for the consumer who needs the emails to use to extract the emails from the commits. Does that make sense?

Reply:
I thought about this approach too, it would require one more foreach to loop for every email. Now thinking about it, obviously it might be better to avoid data duplication

Reply:
What do you mean? Wouldn't it just be an extra loop per gist?

Reply:
You sure? Wouldn't it just be one extra loop per gist? (So not really that many?) I haven't taken a particularly hard look at the code though so I may have missed something crucial! I generally always advocate for code simplicity/easy to understand code over microsecond performance gains.